testsuite/features/support/twopence_init.rb

```diff
@@ -84,14 +84,16 @@
   next if node.nil?
 
   hostname, _local, _remote, code = node.test_and_store_results_together('hostname', 'root', 500)
-  raise "Cannot get hostname for node. Response code: #{code}" if code.nonzero? || hostname.empty?
+  raise "Cannot connect to get client's hostname. Response code: #{code}" if code.nonzero?
```

Comment:
Why are not showing the client again?

I'd add `#{node}` to all messages otherwise we never know about what node the error is.

Reply:
I wasn't exact about the content of node variable before. It is based on FQDN, but it is instance of `Twopence`. Simple said it is egg and chicken problem, we cannot get hostname or FQDN before it is initialized :-/ It was probably reason, why it wasn't here before.

Reply:
Yep, I fall it to the same case sometime ago, I went even to the C code of Twopence to see what it was there, and I couldn't find a way to show here the name.

I'm wondering if Twopence it's really bringing something extra comparing to an easy Ruby class for the remote nodes, like the one I implemented here:
https://github.com/srbarrios/cucumber-ruby-skeleton/blob/master/features/support/remote_node.rb

Reply:
@srbarrios I am glad I am not the only one ;-) I listed all available methods for Twopence and it just wasn't there. I also checked C source, but I still thought I had to miss it I expect something like that must be part of basic Twopence functionality.

Reply:
@srbarrios What do you think about current solution, please? (I just couldn't get it out of my head.) I will understand if you consider it as a strange workaround, but it's works ;-)

Reply:
You are right @lkotek, it's a chicken and egg problem and it's the reason why it was not solved before.